pkg/build/controller/factory/factory.go

```diff
@@ -454,11 +491,15 @@ func (lw *buildPodDeleteLW) List() (runtime.Object, error) {
 			continue
 		}
 		pod, err := lw.KubeClient.Pods(build.Namespace).Get(buildutil.GetBuildPodName(&build))
-		if err != nil && !kerrors.IsNotFound(err) {
-			glog.V(4).Infof("Error getting pod for build %s/%s: %v", build.Namespace, build.Name, err)
-			return nil, err
+		if err != nil {
+			if !kerrors.IsNotFound(err) {
+				glog.V(4).Infof("Error getting pod for build %s/%s: %v", build.Namespace, build.Name, err)
+				return nil, err
+			} else {
+				pod = nil
```

Comment:
Yes it did. Previously the call to `GetBuildLabel` was outside the if condition which might result in the situation @liggitt mentioned. Currently it is inside the else which is ok.